### PR TITLE
#1712: Remove DataObject usage from OpenDialogUrl provider

### LINKED ISSUE #1712: Remove DataObject usage from OpenDialogUrl provider
- Remove constructor of `Magento\Ui\Component\Form\Element\DataType\Media\OpenDialogUrl`
 - Remove corresponding configuration from `app/code/Magento/MediaGalleryIntegration/etc/adminhtml/di.xml`
 - Replace `Magento\MediaGalleryIntegration\Model\OpenDialogUrlProvider` with `Magento\MediaGalleryIntegration\Plugin\NewMediaGalleryOpenDialogUrl` after plugin

Requires discussion before implementation

### PR DESCRIPTION
**Pull request by joweecaquicla**
### Description (*)

This PR will remove DataObject usage from OpenDialogUrl provide, add new plugin and modified the integration test.
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/adobe-stock-integration/issues/1712: Remove DataObject usage from OpenDialogUrl provider

### Manual testing scenarios (*)

Should be tested opening media gallery from wysiwyg, image-uploader (i.e. category image) and page builder
Should be tested for both old and new media gallery

1. Open Media Gallery - the corresponding media gallery is loaded (old or new depending on the configuration)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by joweecaquicla** [comment]
@magento run all tests

**Comment by joweecaquicla** [comment]
@magento run all tests

File: app/code/Magento/Ui/Component/Form/Element/DataType/Media/OpenDialogUrl.php [hunk 1]
```diff
@@ -38,7 +28,7 @@ public function __construct(DataObject $url = null)
     public function get(): string
     {
         if ($this->openDialogUrl) {
```
Comment (sivaschenko) [inline app/code/Magento/Ui/Component/Form/Element/DataType/Media/OpenDialogUrl.php]: Please keep the constructor for ability to initialize `openDialogUrl` property

File: app/code/Magento/MediaGalleryIntegration/composer.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
         "magento/framework": "*",
         "magento/module-media-gallery-ui-api": "*",
         "magento/module-media-gallery-api": "*",
-        "magento/module-media-gallery-synchronization-api": "*"
+        "magento/module-media-gallery-synchronization-api": "*",
+        "magento/module-ui": "*"
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryIntegration/composer.json:10]: Please create a pull request to https://github.com/magento/partners-magento2-infrastructure with an update to composer etalon (see https://github.com/magento/partners-magento2-infrastructure/pull/19 for example) and cross reference the pull requests as related in the description

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @joweecaquicla ! Please see my comments

**Review by sivaschenko — APPROVED** [review]

**Comment by sivaschenko** [comment]
@magento run all tests

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-8010](https://jira.corp.magento.com/browse/ENGCOM-8010) has been created to process this Pull Request
:eight_spoked_asterisk: @sivaschenko, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed
If Enhanced Media Gallery disabled:
Screenshot_105
If Enhanced Media Gallery enabled:
Screenshot_106
All cases are manually covered

**Comment by sivaschenko** [comment]
@magento run all tests

**Pull request merged**